2019-10-27  Prithvi Kannan  <prithvik@lnxsrv03.seas.ucla.edu>

        * timespec.h: Fixed the comparator for timespec_cmp by eliminating overflow
          created by difference and replacing with comparison. First check if one is great          er then check if other is greater, and only if they are equal look at nanosecond          s.

--- timespec.h	2019-10-27 20:09:31.027982000 -0700
+++ timespec.h	2019-10-27 20:10:04.936131000 -0700
@@ -45,8 +45,7 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  return (a.tv_sec > b.tv_sec) ? 1 : (a.tv_sec < b.tv_sec) ? -1 : a.tv_nsec - b.tv_nsec;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
